### DISCUSSION TITLE
Store stage information into meta service

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#5604] [Ideas]
Now we support the `list stage` to show files in the stage after #4472.

Since `list stage` needs `opendal` to call s3 API which is costable, we need to store some information in meta service and synchronize from stage periodically.

Triggers to call list s3 when the following queries happened:
1. `copy into stage`
2. `remove files from stage`
3. `create stage` (external stage may already have files)
4. `drop stage`

Should we remove files in the stage when calling drop stage?

@BohuTANG @junnplus

**Comment by Xuanwo** [comment]
> Should we remove files in the stage when calling drop stage?

`drop stage` and `remove files in stage` are two different actions.

IMHO, we should not remove files while calling drop stage.

**Comment by bohutang** [comment]
I think snow does it well for `drop stage`:
```
For an internal stage, all of the files in the stage are purged from Snowflake, regardless of their load status. 
For an external stage, only the stage itself is dropped; any data files in the referenced external location 
```
https://docs.snowflake.com/en/sql-reference/sql/drop-stage.html

For databend internal stage, it's hard to check a stage file is in loading by another query, so we can force remove all.
